plugins/woocommerce/src/Caches/OrderAggregateCache.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Caches;
+
+use Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore;
+
+/**
+ * A class to cache aggregates from orders.
+ */
+class OrderAggregateCache {
+
+	/**
+	 * Option name.
+	 */
+	const OPTION_NAME = 'woocommerce_orders_aggregate_cache';
+
+	/**
+	 * The running totals for all order statuses.
+	 *
+	 * @var array|null
+	 */
+	private $totals = null;
+
+	/**
+	 * Whether or not the count has changed since loading.
+	 *
+	 * @var bool
+	 */
+	private $has_changed = false;
+
+	/**
+	 * Get the aggregate by identifier.
+	 *
+	 * @param string|string[] $status The order status to count.
+	 * @return int
+	 * @throws \Exception Exception thrown if status it not valid.
+	 */
+	public function get_count( $status ): int {
```

Comment:
With current implementation, we can force the parameter to be `array $status` and force external code to operate in expected data types.

Reply:
Refactored this so that the consumer is responsible for picking statuses they are interested in so we can use the exiting ObjectCache interface.